### PR TITLE
Groups: support for feature flags

### PR DESCRIPTION
**Pull request by macobo**
## Changes

Part of PostHog/posthog#7010

On the API side we only need to pass forward the "groups" argument.

Needed to update `ava` to latest to get tests working.

## Checklist
- [x] Tests for new code (if applicable)
- [x] TypeScript definitions (module.d.ts) updated and in sync with library exports (if applicable)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -52,7 +52,7 @@
         "uuid": "^8.3.2"
     },
     "devDependencies": {
-        "ava": "^0.25.0",
+        "ava": "^3.15.0",
```
Comment (yakkomajuri) [inline package.json:55]: lol

Comment (yakkomajuri) [inline package.json:55]: lol

File: feature-flags.js [hunk 2]
```diff
@@ -61,7 +61,7 @@ class FeatureFlagsPoller {
                 rolloutPercentage: featureFlag.rolloutPercentage,
             })
         } else {
-            const res = await this._request({ path: 'decide', method: 'POST', data: { distinct_id: distinctId } })
+            const res = await this._request({ path: 'decide', method: 'POST', data: { distinct_id: distinctId, groups } })
```
Comment (yakkomajuri) [inline feature-flags.js:64]: super nit: I believe convention is to have constants come first.

**Review by yakkomajuri — APPROVED** [review]
lgtm

File: package.json (hunk 1 above)
Comment (weyert) [reply package.json:55]: cool, I tried to upgrade before but failed miserably. Looks like switching to `require`  might have done the trick?

Comment (macobo) [reply package.json:55]: Yup, there were a few other breaking changes in the library.

**Comment by weyert** [comment]
@macobo Should this work with different rollout percentages?

**Pull request merged**